zipline/pipeline/loaders/blaze/core.py

```diff
@@ -1011,12 +1011,47 @@ def last_in_date_group(df, reindex, have_sids=have_sids):
                     )
                 else:
                     last_in_group = last_in_group.reindex(dates)
+            # Unstack will fill all missing values with NaN; we need to fix
+            # this for all types that are not float.
+            if not df.empty:
+                for column in columns:
+                    if isinstance(column.missing_value, NotSpecifiedType):
+                        raise ValueError("Must specify a missing value for "
+                                         "column %d." % column.name)
+                    # Need special logic in case we're replacing with Nones
+                    elif df[column.name].dtype == categorical_dtype:
+                        last_in_group[column.name] = last_in_group[
+                            column.name
+                        ].where(pd.notnull(last_in_group[column.name]),
+                                column.missing_value)
+                    # Need to convert from float col to datetime col
+                    elif df[column.name].dtype == datetime64ns_dtype:
+                        last_in_group[column.name] = last_in_group[
+                            column.name
+                        ].astype('datetime64[ns]')
+                        last_in_group[column.name] = last_in_group[
+                            column.name
+                        ].fillna(column.missing_value)
+                    else:
+                        last_in_group[column.name] = last_in_group[
+                            column.name
+                        ].fillna(column.missing_value)
 
             return last_in_group
 
         sparse_deltas = last_in_date_group(non_novel_deltas, reindex=False)
         dense_output = last_in_date_group(sparse_output, reindex=True)
-        dense_output.ffill(inplace=True)
+        for column in columns:
+            if have_sids:
+                dense_output[column.name] = dense_output[
+                    column.name
+                ].apply(lambda x: x.replace(
```

Comment:
We can't call `ffill` because it wouldn't work for something like `bool`. How would `replace` work for this?